app/controllers/mixins/generic_form_mixin.rb

```diff
@@ -5,6 +5,7 @@ def flash_and_redirect(*args)
       session[:edit] = nil
       flash_to_session(*args)
       javascript_redirect(previous_breadcrumb_url)
+      session[:flash_msgs] = @flash_array = nil
```

Comment:
This whole `flash_and_redirect` method is supposed to save the flash message in session and redirect,
so that the next request can load it from that session.

Reply:
I know the approach in this PR is not common. But I don't know why "it can't be clearing the session".

Reply:
Because that's the only reason the method exists.

screen 1 - the user does an action that triggers a redirect, and thus can't display the message on screen 1 directly
server - saves the message to session, redirects to screen 2
screen 2 - the new screen loads, and shows the message from session

^^^ that's the reason for the existence of `flash_and_redirect`.

If `flash_and_redirect` clears the session, the user will never see the flash message. That's a bug.